cola/widgets/toolbar.py

```diff
@@ -0,0 +1,527 @@
+# encoding: utf-8
+from __future__ import division, absolute_import, unicode_literals
+
+import json
+
+from cola import cmds
+from cola import guicmds
+from cola.models import prefs
+from cola.qtutils import app
+from cola.widgets import action
+from cola.widgets import compare
+from cola.widgets import createbranch
+from cola.widgets import createtag
+from cola.widgets import editremotes
+from cola.widgets import finder
+from cola.widgets import grep
+from cola.widgets import merge
+from cola.widgets import patch
+from cola.widgets import recent
+from cola.widgets import remote
+from cola.widgets import search
+from cola.widgets import standard
+from cola.widgets import stash
+from qtpy import QtGui
+from qtpy.QtCore import Qt
+from qtpy import QtWidgets
+
+from ..i18n import N_
+from .. import icons
+from .. import qtutils
+from . import defs
+
+
+SOURCE_CONFIG = 'user'
+PREFS_TOOLBAR = 'cola.toolbar'
+SEPARATOR_TEXT = '--------------------'
+
+ACTION_NEW_REPO = 1
+ACTION_OPEN_REPO = 2
+ACTION_OPEN_NEW_WIN = 3
+ACTION_CLONE = 4
+ACTION_REFRESH = 5
+ACTION_FIND_FILES = 6
+ACTION_EDIT_REMOTE_REPO = 7
+ACTION_RECENT_EDIT = 8
+ACTION_APPLY_PATCHS = 9
+ACTION_EXPORT_PATCHS = 10
+ACTION_SAVE_TAR_ZIP = 11
+ACTION_PREFERENCES = 12
+
+ACTION_FETCH = 20
+ACTION_PULL = 21
+ACTION_PUSH = 22
+ACTION_STASH = 23
+ACTION_CREATE_TAG = 24
+ACTION_CHERRY_PICK = 25
+ACTION_MERGE = 26
+ACTION_ABORT_MERGE = 27
+ACTION_RESET_HEAD = 28
+ACTION_RESET_WORKTREE = 29
+ACTION_GREP = 30
+ACTION_SEARCH = 31
+
+ACTION_AMMEND_LAST = 40
+ACTION_STAGE = 41
+ACTION_STAGE_ALL = 42
+ACTION_UNSTAGE_ALL= 43
+ACTION_UNSTAGE = 44
+ACTION_LOAD_COMMIT = 45
+ACTION_APPLY_COMMIT = 46
+
+ACTION_EXPRESSION = 60
+ACTION_BRANCHES = 61
+ACTION_DIFFSTAT = 62
+
+ACTION_REVIEW = 80
+ACTION_CREATE = 81
+ACTION_CHECKOUT = 82
+ACTION_DELETE = 83
+ACTION_DELETE_REMOTE = 84
+ACTION_RENAME = 85
+ACTION_BROWSE_CURRENT = 86
+ACTION_BROWSE_OTHER = 87
+ACTION_VISUALIZE_CURRENT = 88
+ACTION_VISUALIZE_ALL = 89
+
+ACTION_REBASE = 100
+ACTION_REBASE_EDIT = 101
+ACTION_REBASE_CONT = 102
+ACTION_REBASE_IGNORE = 103
+ACTION_REBASE_ABORT = 104
```

Comment:
These IDs should not be changed since they are used to save the toolbar config. I'm thinking now to validate that they exists in TOOLBAR_ACTIONS before inserting the action to prevent errors if we remove one that a user had configured. I also will add a comment about not changing their values.